modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java

```diff
@@ -96,23 +102,29 @@ protected void setupCommandAndEnv() {
 
         withCommand("/bin/bash", "-c", standaloneBaseCommand);
 
+        WaitAllStrategy waitAllStrategy = new WaitAllStrategy();
+
         final String clusterName = getEnvMap().getOrDefault("PULSAR_PREFIX_clusterName", "standalone");
         final String response = String.format("[\"%s\"]", clusterName);
-
-        List<WaitStrategy> waitStrategies = new ArrayList<>();
-        waitStrategies.add(Wait.defaultWaitStrategy());
-        waitStrategies.add(
+        waitAllStrategy.withStrategy(
             Wait.forHttp(ADMIN_CLUSTERS_ENDPOINT).forPort(BROKER_HTTP_PORT).forResponsePredicate(response::equals)
         );
+
         if (transactionsEnabled) {
             withEnv("PULSAR_PREFIX_transactionCoordinatorEnabled", "true");
-            waitStrategies.add(Wait.forHttp(TRANSACTION_TOPIC_ENDPOINT).forStatusCode(200).forPort(BROKER_HTTP_PORT));
+            waitAllStrategy.withStrategy(
+                Wait.forHttp(TRANSACTION_TOPIC_ENDPOINT).forStatusCode(200).forPort(BROKER_HTTP_PORT)
+            );
         }
         if (functionsWorkerEnabled) {
-            waitStrategies.add(Wait.forLogMessage(".*Function worker service started.*", 1));
+            waitAllStrategy.withStrategy(Wait.forLogMessage(".*Function worker service started.*", 1));
         }
-        final WaitAllStrategy compoundedWaitStrategy = new WaitAllStrategy();
-        waitStrategies.forEach(compoundedWaitStrategy::withStrategy);
-        waitingFor(compoundedWaitStrategy);
+
+        if (startupTimeout != null) {
```

Comment:
tbh I am not sure if this is a good strategy.
I would rely more on the health end-point 200 response from `/admin/v2/namespaces/public`

sort of:
```
 Wait.forHttp("/admin/v2/namespaces/public")
                .forStatusCode(200)
                .forPort(PulsarContainer.BROKER_HTTP_PORT)
                .withStartupTimeout(Duration.ofMinutes(3)
```

Reply:
Applying the timeout on the `WaitAllStrategy` is definitely the way to go here. `WaitAllStrategy` already provided different timeout strategies, with `WITH_OUTER_TIMEOUT` being the default.

The implementation of `WaitAllStrategy` is slightly failsafe, meaning `WITH_OUTER_TIMEOUT` will work as expected if you call `withStartupTimeout` *after* adding all strategies, but also working fine when adding strategies *after* `withStartupTimeout` has been applied:
https://github.com/testcontainers/testcontainers-java/blob/de1324ed2800eff4da326d0c23d281399d006bc0/core/src/main/java/org/testcontainers/containers/wait/strategy/WaitAllStrategy.java#L70-L77

Reply:
Thanks for the great feedback and suggestions. PTAL